### DISCUSSION TITLE
Failed some differentiation testset1

### DISCUSSION THREAD

**Discussion by xwaeaewcrhomesysplug** [axkr/symja_android_library#216] [Q&A] [answered]
**Describe the bug**
I did some testing with references from respectable sources.
Below is the Questions used along with answers.

y=((X)^e)^x,find dy/dx.
Ans:((X)^e)^x)*(e^x)*(ln(x)+(1/x))
y=(x)^(1/x),find dy/dx.
Ans:x^(1/x)*((1/(x^2))*(1-ln(x))
y=(Cos(x))^x,x),find dy/dx.
Ans:(Cos(x)^x)*(Ln(Cos(x))-x*Tan(x))

Above pairs should be right mathematically.

Below is output of symja>

D(y=(X^e)^x,x)
(X^e)^x*Log(X^e)

D({y=(x)^(1/x)},x)
x^(1/x)*(1/x^2-Log(x)/x^2)

D(y=(Cos(x))^x,x)
Cos(x)^x*(Log(Cos(x))-x*Tan(x))

As you can see the 1,2 is what I would consider wrong.
With 1,well a part gone?((e^x)*(ln(x)+(1/x)))=Log(X^e)?
2nd,(1-ln(x)=(-Log(x)/x^2)?

A common thing is also,The lack of brackets after a big term with multiplication.
With the absolute no ambigous style,((X)^e)^x)*(
X have power e then power x then that whole obj then multiply other obj.
Symja current,(X^e)^x*
X power e then power x multiply other
case1,power x multiply other
case2,whole object then multiply other(std)

This resolving ambigous in my opinion is lower priority,as long as the symja itself can parses perfectly and does calculation as case2.

**To Reproduce**

Of course,this is math based and thus have math workings steps fo you to refer.
[https://anonymousfiles.io/KWrWSnoI/](url)

I uses the evaluator api you teached me.

**Comment by axkr** [comment, marked as answer]
Please try: `D((x^E)^x,x)`

>> Only the identifiers which consists of only one character are case sensitive. For all other identifiers the input parser doesn't distinguish between lower and upper case characters.

>> For example: the upper-case identifiers D, E, I, N, are different from the identifiers d, e, i, n, whereas the functions like Factorial, Integrate can be entered as factorial(100) or integrate(sin(x),x). If you type SIN(x) or sin(x), Symja assumes you always mean the same built-in Sin function.

See:
- https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/01-getting-started.md

`Ln` is only an alias for easier input:
- https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/functions/Log.md
- https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/functions/Ln.md

**Comment by xwaeaewcrhomesysplug** [comment]
Thank you very much,it worked .Just a extra question where can I access the constants?
BTW,what is the condition for them to show Ln instead of log?(just kinda curious)
(E^x/x+E^x*Log(x))>(e^x)(ln(x)+(1/x))

**Comment by xwaeaewcrhomesysplug** [comment]
will do more test on my own too.if have a full report will inform you.^^
